lib/rubycritic/commands/compare.rb

```diff
@@ -0,0 +1,115 @@
+# frozen_string_literal: true
+require 'rubycritic/source_control_systems/base'
+require 'rubycritic/analysers_runner'
+require 'rubycritic/revision_comparator'
+require 'rubycritic/reporter'
+require 'rubycritic/commands/base'
+require 'rubycritic/commands/default'
+
+module RubyCritic
+  module Command
+    class Compare < Default
+      def initialize(options)
+        super
+        @build_number = 0
+      end
+
+      def execute
+        compare_branches
+        status_reporter.score = Config.feature_branch_score
+        status_reporter
+      end
+
+      private
+
+      attr_reader :paths, :status_reporter
+
+      def compare_branches
+        update_build_number
+        set_root_paths
+        Config.no_browser = true
+        analyse_branch(:base_branch)
+        analyse_branch(:feature_branch)
+        Config.no_browser = false
+        analyse_modified_files
+        compare_code_quality
+      end
+
+      # keep track of the number of builds and
+      # use this build number to create seperate directory for each build
+      def update_build_number
+        build_file_location = '/tmp/build_count.txt'
+        File.new(build_file_location, 'a') unless File.exist?(build_file_location)
+        @build_number = File.open(build_file_location).readlines.first.to_i + 1
+        File.write(build_file_location, @build_number)
+      end
+
+      def set_root_paths
+        Config.base_root_directory = Pathname.new(branch_directory(:base_branch))
+        Config.feature_root_directory = Pathname.new(branch_directory(:feature_branch))
+        Config.build_root_directory = Pathname.new(build_directory)
+      end
+
+      # switch branch and analyse files
+      def analyse_branch(branch)
+        SourceControlSystem::Git.switch_branch(Config.send(branch))
+        critic = critique(branch)
+        Config.send(:"#{branch}_score=", critic.score)
+        Config.root = branch_directory(branch)
```

Comment:
I don't think we should be overriding the arguments. At least make a copy and revert it after this run.

In fact, I'm thinking of removing the attr_writer methods for all the options. :thinking: 
If we get a bug related to the options, it'll be hell to track it down because we're changing them mid analysis.